### PR TITLE
Add notification for closeAccount

### PR DESCRIPTION
**Pull request by AlexanderWillner**
We've already all required hooks and code for sending a notification when a member gets activated. This patch adds the same functionality for the `closeAccount` event.

This change was implemented and tested using `1.6.14`. For the pull request a diff was generated for `main`. A backport for `1.6.15` would be nice.

### CODE REVIEW

**Comment by fritzmg** [comment]
> A backport for `1.6.15` would be nice.

I am pretty sure `terminal42/notification_center` uses semantic versioning. New feature would go into `1.8`. Why do you need it for `1.6`?

File: library/NotificationCenter/ContaoHelper.php [hunk 1]
```diff
@@ -87,6 +87,23 @@ public function sendActivationEmail($objMember, &$objModule)
...
+     *
+     * @param object
+     * @param object
+     */
+    public function sendCloseEmail($objMember, $mode, &$objModule)
+    {
+        if (!$objModule->nc_close_notification) {
+            return;
+        }
+        $objMember = \MemberModel::findByPk($objMember);
```
Comment (fritzmg) [inline library/NotificationCenter/ContaoHelper.php:101]:
```suggestion
        }

        $objMember = MemberModel::findByPk($objMember);
```

File: languages/de/tl_nc_notification.xlf [hunk 2]
```diff
@@ -130,6 +130,11 @@
         <source>Member activation</source>
         <target>Mitglied: Aktivierung</target>
         
+      </trans-unit>
+      <trans-unit id="tl_nc_notification.type.member_close.0">
+        <source>Member closing</source>
+        <target>Mitglied: Schließung</target>
+
```
Comment (fritzmg) [inline languages/de/tl_nc_notification.xlf:137]: Translations are managed on [Transifex](https://explore.transifex.com/terminal42/notification_center/) - you only need to provide the English translations and remove the other languages.

File: languages/de/tl_module.xlf [hunk 3]
```diff
@@ -20,6 +20,16 @@
...
+      </trans-unit>
+      <trans-unit id="tl_module.nc_close_notification.0">
+        <source>Close notification</source>
+        <target>Schließungsbenachrichtigung</target>
+        
+      </trans-unit>
+      <trans-unit id="tl_module.nc_close_notification.1">
+        <source>Please select a close notification.</source>
+        <target>Bitte wählen Sie eine Schließungsbenachrichtigung.</target>
+
```
Comment (fritzmg) [inline languages/de/tl_module.xlf:32]: Translations are managed on [Transifex](https://explore.transifex.com/terminal42/notification_center/) - you only need to provide the English translations and remove the other languages.

File: dca/tl_module.php [hunk 4]
```diff
@@ -54,3 +57,4 @@
 $GLOBALS['TL_DCA']['tl_module']['fields']['nc_notification']['eval']['ncNotificationChoices']['newsletterActivateNotificationCenter'] = array('newsletter_activate');
 $GLOBALS['TL_DCA']['tl_module']['fields']['nc_notification']['eval']['ncNotificationChoices']['newsletterUnsubscribeNotificationCenter'] = array('newsletter_unsubscribe');
 $GLOBALS['TL_DCA']['tl_module']['fields']['nc_activation_notification']['eval']['ncNotificationChoices']['registration'] = array('member_activation');
+$GLOBALS['TL_DCA']['tl_module']['fields']['nc_close_notification']['eval']['ncNotificationChoices']['registration'] = array('member_close');
```
Comment (fritzmg) [inline dca/tl_module.php:60]:
```suggestion
$GLOBALS['TL_DCA']['tl_module']['fields']['nc_close_notification']['eval']['ncNotificationChoices']['registration'] = array('member_close');

```

**Review by fritzmg — CHANGES_REQUESTED** [review]

**Comment by fritzmg** [comment]
> > A backport for 1.6.15 would be nice.
>
> I am pretty sure terminal42/notification_center uses semantic versioning. New feature would go into 1.8. Why do you need it for 1.6?

Btw. the changes in this PR could easily be integrated as an App customisation or even extension of your own, if you really need it for an older feature version.

**Comment by AlexanderWillner** [comment]
> > I am pretty sure terminal42/notification_center uses semantic versioning. New feature would go into 1.8. Why do you need it for 1.6?

As version 1.6 is needed for setups with Haste Version 4.*

> Btw. the changes in this PR could easily be integrated as an App customisation or even extension of your own, if you really need it for an older feature version.

We've many customisations, just trying to give changes back to the community.

**Review by fritzmg — APPROVED** [review]
lgtm in general :)

Two things to consider though:

The name of the new `member_close_mode` token does not make much sense to me. The `member_*` tokens are supposed to represent the properties of the member itself. However the `member_close_mode` represents the `reg_close` setting in the close account module. imho the token's name should be just `close_mode` for example.

The new token is also missing in the `$GLOBALS['NOTIFICATION_CENTER']['NOTIFICATION_TYPE']` array. It should be added to `email_subject`, `email_text` and `email_html` for the `member_close` notification type.

**Comment by fritzmg** [comment]
👍 

The `tokens.xlf` still needs to be updated though with a description of thew new token.

**Comment by AlexanderWillner** [comment]
Added a suggestion.

**Comment by fritzmg** [comment]
Did you test this change? Doesn't it need to be `NOTIFICATION_CENTER_TOKEN.member_close.close_mode`?

**Comment by Toflar** [comment]
Thanks for the contribution! However, I won't be releasing any new 1.x version anymore. Even if I did, it would not be helpful to you as it would have to be 1.8 and thus haste 5-only anyway - I don't want to backport anything and release as bugfix release and potentially causing other issues unless it's absolutely necessary (which a new feature does not qualify for).

I would love to see the PR being picked up again after v2 is released, though. Maybe it can be a nice addition to the core then and be part of a future 2.1 or 2.x release? :) 

In the meantime, feel free to publish this as your own extension which allows you to meet all your needs.

**Comment by Toflar** [comment]
I'm cleaning up issues and PRs so I will have to close this PR but please, feel free to work on a PR for version 2 (to be released very, very soon) once it's here if you still care 😊

**Pull request closed without merging**